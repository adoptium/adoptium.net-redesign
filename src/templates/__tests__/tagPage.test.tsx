import React from "react"
import { render } from "@testing-library/react"
import TagPage, { Head } from "../tagPage"
import { describe, expect, it } from "vitest"
import { axe } from "vitest-axe"
import { createMDXData } from "../../__fixtures__/page"

let mockData = createMDXData()
const pageContext = {
  tag: "test",
  previousPageNumber: null,
  nextPageNumber: 2,
  currentPageNumber: 1,
  numTagPages: 5,
}

describe("TagPage pages", () => {
  it("renders correctly", () => {
    const { container } = render(
      <TagPage data={mockData} pageContext={pageContext} />,
    )
    // eslint-disable-next-line
    const pageContent = container.querySelector("main")

    expect(pageContent).toMatchSnapshot()
  })

  it("renders correctly - with next pagination", () => {
    const { container } = render(
      <TagPage
        data={mockData}
        pageContext={{ ...pageContext, nextPageNumber: 2 }}
      />,
    )
    // eslint-disable-next-line
    const pageContent = container.querySelector("main")

    expect(pageContent).toMatchSnapshot()
  })

  it("renders correctly - with previous pagination", () => {
    const { container } = render(
      <TagPage
        data={mockData}
        pageContext={{ ...pageContext, previousPageNumber: 1 }}
      />,
    )
    // eslint-disable-next-line
    const pageContent = container.querySelector("main")
    expect(pageContent).toMatchSnapshot()
  })

  it("renders correctly - with next and previous pagination", () => {
    const { container } = render(
      <TagPage
        data={mockData}
        pageContext={{
          ...pageContext,
          nextPageNumber: 2,
          previousPageNumber: 1,
        }}
      />,
    )
    // eslint-disable-next-line
    const pageContent = container.querySelector("main")
    expect(pageContent).toMatchSnapshot()
  })

  it("head renders correctly", () => {
    const { container } = render(<Head pageContext={pageContext} />)
    // eslint-disable-next-line
    const title = container.querySelector("title")
    expect(title?.textContent).toEqual("test | Adoptium")
  })

  it("has no accessibility violations", async () => {
    const { container } = render(
      <TagPage data={mockData} pageContext={pageContext} />,
    )
    const results = await axe(container)
    expect(results).toHaveNoViolations()
  })
})
