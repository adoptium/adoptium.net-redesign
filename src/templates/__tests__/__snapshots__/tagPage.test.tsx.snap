// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TagPage pages > renders correctly 1`] = `
<main>
  <section
    class="py-5 container"
  >
    <div
      class="row py-lg-5"
    >
      <div
        class="col-lg-9 col-md-9 mx-auto"
      >
        <h1>
          test
        </h1>
        <hr
          class="pb-5"
        />
        <article
          class="pb-3"
        >
          <header>
            <h2
              class="h3"
            >
              <a
                href="/mdx/mdx-page-title"
              >
                MDX Page title
              </a>
            </h2>
            <div
              class="flex justify-center items-center gap-5"
            >
              <img
                alt="Adoptium PMC"
                image="[object Object]"
                stlye="[object Object]"
                style="margin-left: 0.5rem; margin-bottom: 0px; min-width: 30px; border-radius: 100%;"
              />
              <span
                class="text-[16px] font-bold leading-[150%] text-white"
              >
                <a
                  href="/blog/author/pmc"
                >
                  <span
                    class="text-primary pr-2"
                  >
                    Adoptium PMC
                    ,
                  </span>
                   
                  2021-01-01
                </a>
              </span>
            </div>
          </header>
          <section>
            <p>
              MDX Page excerpt
               
              <a
                href="/mdx/mdx-page-title"
              >
                Read more
              </a>
            </p>
          </section>
        </article>
      </div>
    </div>
  </section>
</main>
`;
